components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/AssetBlob.java

```diff
@@ -35,6 +36,8 @@
 {
   private final BlobRef blobRef;
 
+  private final Blob blob;
```

Comment:
@mcculls I didn't like this much when I made it, but better suggestions didn't leap to mind at the time. The overlap between Blob and BlobRef is just the BlobId - I suppose I could tease the AssetBlob.blobRef apart and instead store Node/Store names, then reconstitute a blobRef in the getter, but it seems like a limited gain.

I think @cstamas is on the right track: this oddity exists because there's not currently a clean service inside Nexus where you can just get(BlobRef) to obtain bytes from whatever Nexus node. (This might return something Blob-like, from which you could call .getBlobRef().) Since that abstraction fits between StorageTxImpl and BlobStore, this code will get revisited then.